### PR TITLE
docs: bring the spec to the spec repo

### PR DESCRIPTION
**Pull request by daviddias**
//cc @stebalien

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,83 @@
...
+
+## Message format
+
+Every communication in multiplex consists of a header, and a length prefixed data segment.
+
+The header is an unsigned base128 varint, as defined in the [protocol buffers spec](https://developers.google.com/protocol-buffers/docs/encoding#varints). The lower three bits are the message flags, and the rest of the bits (shifted down by three bits) are the stream ID this message pertains to:
+
+```
+header = readUvarint()
+flag = head & 0x07
```
Comment (richardschneider) [inline README.md:25]: Should be `header & 0x07`

File: README.md [hunk 2]
```diff
@@ -1,2 +1,83 @@
...
+id = flag >> 3
```
Comment (richardschneider) [inline README.md:26]: should `header >> 3`

File: README.md [hunk 3]
```diff
@@ -1,2 +1,83 @@
...
+
+### Flag Values
+
+| NewStream        | 0 |
+| MessageReceiver  | 1 |
+| MessageInitiator | 2 |
+| CloseReceiver    | 3 |
+| CloseInitiator   | 4 |
+| ResetReceiver    | 5 |
+| ResetInitiator   | 6 |
```
Comment (richardschneider) [inline README.md:37]: Not displaying correctly.  Try adding a header to the table, e.g. `| Flag | Value| \n | ---- | ------ |`

File: README.md [hunk 4]
```diff
@@ -1,2 +1,83 @@
...
+
+The data segment is length prefixed by another unsigned varint. This results in one message looking like:
+
+| header  | length  | data           |
```
Comment (richardschneider) [inline README.md:41]: Again a header with `| ---- | ---- | ---- |`

File: README.md (hunk 3 above)
Comment (daviddias) [reply README.md:37]: I fixed it in master. https://github.com/libp2p/mplex/commit/ca1d70a37df90fcf522479a5afa873309ccf67ac

File: README.md (hunk 1 above)
Comment (daviddias) [reply README.md:25]: Thanks for the review @richardschneider. This was a straight c&p from @Stebalien's notes. Mind submitting a PR with the proposed fixes?

File: README.md (hunk 4 above)
Comment (daviddias) [reply README.md:41]: Fixed in master https://github.com/libp2p/mplex/commit/ca1d70a37df90fcf522479a5afa873309ccf67ac

**Pull request merged**